std/encoding.d

```diff
@@ -3380,3 +3380,111 @@ version(unittest)
         return "0123456789ABCDEF"[n & 0xF];
     }
 }
+
+import std.typecons : tuple, Tuple;
+import std.conv : to;
+
+/** Definitions of common Byte Order Marks
+*/
+enum BOM
+{
+    noBOM     = 0,  /// no BOM was found
+    utf32be   = 1,  /// [0x00, 0x00, 0xFE, 0xFF]
+    utf32le   = 2,  /// [0xFF, 0xFE, 0x00, 0x00]
+    utf7      = 3,  /*  [0x2B, 0x2F, 0x76, 0x38]
+                        [0x2B, 0x2F, 0x76, 0x39],
+                        [0x2B, 0x2F, 0x76, 0x2B],
+                        [0x2B, 0x2F, 0x76, 0x2F],
+                        [0x2B, 0x2F, 0x76, 0x38, 0x2D]
```

Comment:
Right, skipping 4 bytes for non-empty UTF-7 files is a no-go. But if I understand it correctly, the 5 byte signature is special for empty files. If that's the case, then it would be a mistake to interpret it as a file with 10 bits of content (2 bits from the signature leftover, 8 bits for the terminating `0x2D`).

I guess nobody is surprised that UTF-7 doesn't jive well in a common API with other encodings...